components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManagementService.java

```diff
@@ -103,7 +105,127 @@ public void updateResidentIdP(IdentityProvider identityProvider)
+     *
+     * @return total count of Idps
+     * @throws IdentityProviderManagementException Throw exception while get the count of Idps.
+     */
+    public int getAllIdpCount() throws IdentityProviderManagementException {
+
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        return IdentityProviderManager.getInstance().getTotalIdPCount(IdPManagementConstants.EMPTY_STRING,
+                tenantDomain);
+    }
+
+    /**
+     * Retrieves registered Identity providers for the logged-in tenant.
+     *
+     * @return Array of <code>IdentityProvider</code>. IdP names, primary IdP and home.
+     * realm identifiers of each IdP.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    public IdentityProvider[] getAllPaginatedIdpInfo(int pageNumber)
+            throws IdentityProviderManagementException {
+
+        return getPaginatedIdpInfo(IdPManagementConstants.EMPTY_STRING, pageNumber);
+    }
+
+    /**
+     * Retrieves Identity providers list of array for the logged-in tenant.
+     *
+     * @param filter     searching value.
+     * @param pageNumber page number.
+     * @return Identity providers list of array.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    public IdentityProvider[] getPaginatedIdpInfo(String filter, int pageNumber)
+            throws IdentityProviderManagementException {
+
+        validateRequestedPageNumber(pageNumber);
+        Integer limit = getIdpPageLimit();
+        Integer offset = getIdpPageOffset(pageNumber, limit);
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            IdpSearchResult identityProviderList = IdentityProviderManager.getInstance()
+                    .getIdPs(limit, offset, filter, IdPManagementConstants.DEFAULT_SORT_ORDER,
+                            IdPManagementConstants.DEFAULT_SORT_BY, tenantDomain);
+            return identityProviderList.getIdPs().toArray(new IdentityProvider[0]);
+        } catch (IdentityProviderManagementException idpException) {
+            log.error("Error while getting IdPs in tenantDomain : " + tenantDomain, idpException);
+            throw idpException;
+        }
+    }
+
+    /**
+     * Retrieves Identity providers list of array for the logged-in tenant.
+     *
+     * @param filter searching value.
+     * @return Identity providers list of array.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    public int getFilteredIdpCount(String filter)
+            throws IdentityProviderManagementException {
+
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        return IdentityProviderManager.getInstance()
+                .getTotalIdPCount(IdPManagementConstants.EMPTY_STRING, tenantDomain);
```

Comment:
Why are we passing "IdPManagementConstants.EMPTY_STRING" here? Isn't this API written to get the total filtered IdP count?

Reply:
yes, We introduce new API (getTotalIdPCount) for getting paginated and total idp count. We need this to display the idp list in the management console.

Reply:
Then what is the purpose of having getAllIdpCount() API here? I see both getAllIdpCount() and getFilteredIdpCount() APIs have the same implementation. IMO getAllIdpCount() should be for getting the total count of IdPs whereas getFilteredIdpCount() should be for getting the total count of IdPs for a given filter. Hence shouldn't this method be returning the following? 

`return IdentityProviderManager.getInstance().getTotalIdPCount(filter, tenantDomain);`